reactive/swarm.py

```diff
@@ -25,9 +26,17 @@ def swarm_etcd_cluster_setup(etcd):
     """
     bind_docker_daemon()
     con_string = etcd.connection_string().replace('http', 'etcd')
-    start_swarm_etcd_agent(con_string)
-    if hookenv.is_leader():
-        start_swarm_etcd_manager(con_string)
+
+    opts = {}
+    opts['connection_string'] = con_string
+    opts['addr'] = hookenv.unit_private_ip()
+    opts['port'] = 2375
+    opts['leader'] = is_leader()
```

Comment:
This is a very clever how you use 'leader' in the template.  If we ever get true leadership from Juju (leader dies, new one is selected) will this code work properly?